sdk/formrecognizer/ai-form-recognizer/src/documentModelAdministrationClient.ts

```diff
@@ -0,0 +1,613 @@
+   *       fieldSchema,
+   *       // the service's confidences in the fields (an object with field names as properties and numeric confidence
+   *       // values)
+   *       fieldConfidence
+   *     }
+   *   }
+   * } = await client.getModel(modelId);
+   * ```
+   *
+   * @param modelId - the unique ID of the model to query
+   * @param options - optional settings for the request
+   * @returns information about the model with the given ID
+   */
+  public getModel(modelId: string, options?: GetModelOptions): Promise<ModelInfo> {
+    return this._restClient.getModel(modelId, options);
+  }
+
+  /**
+   * List summaries of models in the resource. Custom as well as prebuilt models will be included. This operation
+   * supports paging.
+   *
+   * The model summary ({@link ModelSummary}) includes only the basic information about the model, and does not include
+   * information about the document types in the model (such as the field schemas and confidence values).
+   *
+   * To access the full information about the model, use {@link getModel}.
+   *
+   * ### **Breaking Change**
+   *
+   * In previous versions of the Form Recognizer REST API and SDK, the `listModels` method would return all models, even
+   * those that failed to create due to errors. In the new service versions, `listModels` and `getModels` _only produce
+   * successfully created models_ (i.e. models that are "ready" for use). Failed models are now retrieved through the
+   * "operations" APIs, see {@link getOperation} and {@link listOperations}.
+   *
+   * ### Examples
+   *
+   * #### Async Iteration
+   *
+   * ```javascript
+   * for await (const summary of client.listModels()) {
+   *   const {
+   *     modelId, // The model's unique ID
+   *     description, // a textual description of the model, if provided during model creation
+   *   } = summary;
+   *
+   *   // You can get the full model info using `getModel`
+   *   const model = await client.getModel(modelId);
+   * }
+   * ```
+   *
+   * #### By Page
+   *
+   * ```javascript
+   * // The listModels method is paged, and you can iterate by page using the `byPage` method.
+   * const pages = client.listModels().byPage();
+   *
+   * for await (const page of pages) {
+   *   // Each page is an array of models and can be iterated synchronously
+   *   for (const model of page) {
+   *     const {
+   *       modelId, // The model's unique ID
+   *       description, // a textual description of the model, if provided during model creation
+   *     } = summary;
+   *
+   *     // You can get the full model info using `getModel`
+   *     const model = await client.getModel(modelId);
+   *   }
+   * }
+   * ```
+   *
+   * @param options - optional settings for the model requests
+   * @returns an async iterable of model summaries that supports paging
+   */
+  public listModels(options?: ListModelsOptions): PagedAsyncIterableIterator<ModelSummary> {
+    return this._restClient.listModels(options);
```

Comment:
In case you need to massage the result object based on feedback here, you can use [`getPagedAsyncIterator`](https://github.com/Azure/azure-sdk-for-js/blob/497cfdedc5d01e1c20dbeb51050af7e9ebca9aeb/sdk/core/core-paging/src/getPagedAsyncIterator.ts#L13)